scipy/optimize/_differentiable_functions.py

```diff
@@ -587,10 +603,11 @@ def jac_dot_v(x, v):
 
             def update_hess():
                 self._update_jac()
-                self.H = approx_derivative(jac_dot_v, self.x,
-                                           f0=self.J.T.dot(self.v),
-                                           args=(self.v,),
-                                           **finite_diff_options)
+                self.H, dct = approx_derivative(jac_dot_v, self.x,
```

Comment:
`jac_dot_v` doesn't need to be pickleable because `finite_diff_options['as_linear_operator'] is True`. 

The creation of the `LinearOperator` by `approx_derivative` doesn't consume any function evaluations, they are only consumed when the `LinearOperator` is used. Each call of the `LinearOperator` will use two or three evaluations, which didn't seem worth parallelising to me. If it's not parallelised, then `jac_dot_v` doesn't need to be pickleable.